test/spec/upload.browser.js

```diff
@@ -171,5 +173,510 @@ describe("tus", function () {
       expect(err.message).toBe("tus: unexpected response while creating upload, originated from request (response code: 500, response text: server_error)");
       expect(err.originalRequest).toBeDefined();
     });
+
+    it("should upload data from a reader", function (done) {
+      var reader = {
+        value: "hello world".split(""),
+        read: function () {
+          if (this.value) {
+            const value = this.value;
+            this.value = null;
+            return Promise.resolve({ value, done: false });
+          } else {
+            return Promise.resolve({ value: undefined, done: true });
+          }
+        }
+      };
+
+      var options = {
+        endpoint: "http://tus.io/uploads",
+        headers: {
```

Comment:
It's not necessary to assert custom headers, metadata and withCredentials in every test case. This just makes it harder to read. Would you mind removing them from your new test cases?

Reply:
You're totally right. I removed a lot of clutter!